src/AdminApp/ManageSalons.js

```diff
@@ -0,0 +1,249 @@
+        }
+
+        console.log(json);
+        console.log("Salons i category i bounds");
+      }
+    }
+    load();
+  }, [salons]);
+
+  useEffect(() => {
+    async function load() {
+      if (category !== null && bbox !== null && salons.length === 0) {
+        const response = await fetch(
+          `http://localhost:8080/services/${category}`,
+          {
+            method: "POST",
+            body: JSON.stringify({
+              serviceName: category,
+              LatBtm: bbox.btmRightPoint?.lat,
+              LonBtm: bbox.btmRightPoint?.lon,
+              LatTop: bbox.topLeftPoint?.lat,
+              LonTop: bbox.topLeftPoint?.lon,
+              distance: 0,
+              treatment: null,
+              priceFrom: 0,
+              priceTo: 0,
+              opinion: 0.0,
+              localizationLat: latitude,
+              localizationLon: longitude,
+            }),
+            headers: {
+              "Content-Type": "application/json",
+            },
+          }
+        );
+        let json = [];
+
+        if (response.ok) {
+          json = await response.json();
+          setSalons(json);
+        } else {
+          alert("Nie znaleziono rekordow  w bazie danych dla " + category);
+        }
+
+        console.log(salons.length);
+        console.log("Salons i category i bbox");
+      }
+    }
+    load();
+  }, [salons]);
+
+  useEffect(() => {
+    function success(position) {
+      setLatitude(position.coords.latitude);
+      setLongitude(position.coords.longitude);
+    }
+
+    function errorF() {
+      setError("Unable to retrieve your location");
+    }
+
+    navigator.geolocation.getCurrentPosition(success, errorF);
+  }, []);
+
+  return (
+    <>
+      <div className="App" style={{ height: "100vh" }}>
+        <div className="MapContainer">
+          <Menu setCategory={setCategory} />
+          <Map
+            mapStyle="https://api.tomtom.com/map/1/style/20.0.1-1/basic_main-lite.json?key=1ncwaIygtJ0KrjH5ssohlEKUGFf7G5Dv"
+            mapProps={{ center: [100.0, 0], zoom: 10 }}
+          >
+            <Markers salons={salons} setSalons={setSalons} />
+
+            <SearchAndZoom
+              searchConfig={{ limit: 40 }}
+              highlightSuggestions={true}
+              emptyState={<div style={{ color: "red" }}>{"Nothing found"}</div>}
+              position="top-right"
+              renderMarker={true}
+              onSearch={(item) => {
+                setBbox(item.bbox);
+              }}
+              isAutoFocusActive={true}
+            />
+
+            <ZoomToLocalization
+              latitude={latitude}
+              longitude={longitude}
+              setBounds={setBounds}
+            />
+          </Map>
+        </div>
+      </div>
+    </>
+  );
+};
+
+function ZoomToLocalization({ latitude, longitude, setBounds }) {
+  const { map } = useMap();
+  useEffect(() => {
+    map.panTo([longitude, latitude]);
+    setTimeout(() => {
```

Comment:
To avoid timeout you can use `map.flyTo` (https://docs.mapbox.com/mapbox-gl-js/api/map/#map#flyto) instead of the `map.panTo`